### ISSUE TITLE
PP-ct webform doesn't load Paypal checkout

### ISSUE THREAD

**Issue by eileenmcnaughton** [eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor#107]
Webform isn't loading the script :-(

**Comment by mattwire** [comment]
That's why I load it via alterContent in stripe :-( https://lab.civicrm.org/extensions/stripe/blob/master/stripe.php#L159

**Comment by eileenmcnaughton** [comment]
@mattwire yeah - but we need webform to interact with our payment infrastructure in a way that doesn't require alterContent IMHO

**Comment by mattwire** [comment]
Agreed - it's pretty hacky doing it that way - but currently no other choice.  I wonder if some kind of standard hook/function that both webform and civi calls would work?  As this kind of processor loading specific resources is becoming more common.

**Comment by eileenmcnaughton** [comment]
Right - I'm asking @colemanw to help out here - we have the buildForm function on CRM_Core_Payment for this purpose but for unknown reasons webform doesn't call it or it doesn't work there

**Comment by joemcl** [comment]
hi @colemanw @eileenmcnaughton any update on this, and see 
https://civicrm.stackexchange.com/questions/34058/can-omnipay-paypal-checkout-be-used-with-membership-signups-contributions?noredirect=1#comment39184_34058

**Comment by eagleuk** [comment]
It's definitely calling the buildForm function as far as I can tell.

However it seems like adding vars doesn't seem to work within this function.

` CRM_Core_Resources::singleton()->addVars('omnipay', $jsVariables);`  has no effect.

However it doesn't look like there is actually anything in the vars array on the page. 

Therefore the paypalScriptURL cannot be set, even if you fixed that you'll end up hitting issues in the createBillingAgreement function before those vars don't exist either.

`var paypalScriptURL = 'https://www.paypal.com/sdk/js?client-id=' + CRM.vars.omnipay.client_id + '&currency=' + CRM.vars.omnipay.currency + '&commit=false&vault=true';`

Will investigate further and see what I can find but inexperienced with this functionality.

**Comment by eagleuk** [comment]
Look like @mattwire  May have added more stuff in the [Stripe extension to work around this](https://lab.civicrm.org/extensions/stripe/commit/a815fc5b88bf3a7b584df71ab7c3df193995ce76?view=parallel#8c91cd6e02da2e1f55014fa9d586f1af83e8e96f).

**Comment by mattwire** [comment]
@eagleuk Yes :-) We do: https://lab.civicrm.org/extensions/stripe/blob/master/templates/CRM/Core/Payment/Stripe/Card.tpl#L3-15

**Comment by eagleuk** [comment]
@mattwire Thanks, I'll see what I can figure out but as someone a bit more inexperienced I may struggle, I lack any background knowledge :D.

**Comment by eagleuk** [comment]
@mattwire Could you please point me ( if possible ) in the right direction of what to do once the current onApprove function runs? 

Currently it will call the form.submit(); function, which I believe to be incorrect at this stage and redirect to the start of the process, without the confirmation step to approve payment etc. or submitting the form. 

I've taken a look at your Stripe work and haven't worked out the correct steps to take yet, seems like there is going to be a number of things to add to handle the submission process?
